src/logger.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <locale>
+#include <sstream>
+
+namespace ohl::logger {
+
+
+/**
+ * @brief Logs a logMessage to the log file ohl::logger::logPath. \n will be
+ * added at the end
+ *
+ * @param logMessage the message to be logged.
+ */
+void log( const char * logMessage );
+void log( const wchar_t * logMessage );
+void log( const std::ostream& logMessage );
+void log( const std::wostream& logMessage );
```

Comment:
I get how you arrived at them, but these signatures aren't really that useful, you still need to construct the strings yourself.

There are really two approaches to c++ logging libraries:
1. Stream based, where you export an `ostream` object, and users do
   ```cpp
   log << L"[OHL] Failed to open file '" << path << L"'!";
   ```
2. Varargs based, which take a printf or fmtlib style format string and a bunch of arguments.
   ```cpp
   log("[OHL] Failed to open file '%s'!", path);
   ```

I don't particularly care which of the two is used, as long as you don't have to construct strings yourself.